### PR TITLE
RESOURCE-424 AWS::ApiGatewayV2::Integration

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+WHEN_NO_MATCH passes the request body for unmapped content types through to the integration backend without transformation.
+NEVER rejects unmapped content types with an HTTP 415 Unsupported Media Type response.
+WHEN_NO_TEMPLATES allows pass-through when the integration has no content types mapped to templates. However, if there is at least one content type defined, unmapped content types will be rejected with the same HTTP 415 Unsupported Media Type response.
+
+`payload_format_version`
+: Specifies the format of the payload sent to an integration. Required for HTTP APIs.
+
+`request_parameters`
+: For WebSocket APIs, a key-value map specifying request parameters that are passed from the method request to the backend. The key is an integration request parameter name and the associated value is a method request parameter value or static value that must be enclosed within single quotes and pre-encoded as required by the backend. The method request parameter value must match the pattern of method.request.{location}.{name}, where {location}is querystring, path, or header; and {name}must be a valid and unique method request parameter name.
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration.md:98]: A doubt - dont we have to mention abour response parameter?

**Review by dkumaras — COMMENTED** [review]
Please check and confirm for approving from doc side.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration.md (hunk 1 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration.md:107]: Yeah... I have added it. Please check.

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_v2_integration.md:107]: ok

**Review by dkumaras — APPROVED** [review]

**Pull request merged**